modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/utils/DeployResultFileUtils.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.modules.deployment.utils;
+
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.channels.FileChannel;
+import java.nio.channels.FileLock;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import org.awaitility.Awaitility;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * Tool class for files after deployment is completed.
+ */
+@Component
+public class DeployResultFileUtils {
+
+    private static final String MODE = "rw";
+    @Value("${wait.time.for.deploy.result.file.lock.in.seconds}")
+    private int awaitAtMost;
+    @Value("${polling.interval.for.deploy.result.file.lock.check.in.seconds}")
+    private int awaitPollingInterval;
+
+    /**
+     * Method to await for the tfstate file to be unoccupied.
+     */
+    public void awaitFileNotLocked(String path) {
+        Awaitility.await()
+                .atMost(Duration.ofSeconds(awaitAtMost))
+                .pollInterval(Duration.ofSeconds(awaitPollingInterval))
+                .pollDelay(0, TimeUnit.SECONDS)
+                .until(() -> !isTfStateFileLocked(path));
```

Comment:
we need to also catch the exception like this
https://github.com/eclipse-xpanse/xpanse/blob/main/modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/polling/StatusChangePolling.java#L97